core/domain/user_jobs_one_off_test.py

```diff
@@ -2869,3 +2869,326 @@ def test_draft_discarded_if_it_is_too_old(self):
+            [self.EXP_2_ID]
+        )
+        self.assertEqual(
+            user_models.IncompleteActivitiesModel.get(
+                self.USER_3_ID
+            ).exploration_ids,
+            []
+        )
+        self.assertEqual(
+            user_models.UserSubscriptionsModel.get(
+                self.USER_1_ID
+            ).exploration_ids,
+            [self.EXP_2_ID]
+        )
+        self.assertEqual(
+            user_models.UserSubscriptionsModel.get(
+                self.USER_3_ID
+            ).exploration_ids,
+            [self.EXP_2_ID]
+        )
+
+
+class DeleteNonExistentExpUserDataJobTests(test_utils.GenericTestBase):
+
+    USER_1_ID = 'user_1_id'
+    USER_2_ID = 'user_2_id'
+    EXP_1_ID = 'exp_1_id'
+    EXP_2_ID = 'exp_2_id'
+
+    def setUp(self):
+        super(DeleteNonExistentExpUserDataJobTests, self).setUp()
+        self.signup(self.OWNER_EMAIL, self.OWNER_USERNAME)
+        self.owner_id = self.get_user_id_from_email(self.OWNER_EMAIL)
+        self.owner = user_services.get_user_actions_info(self.owner_id)
+        self.save_new_valid_exploration(self.EXP_1_ID, self.owner_id)
+        self.save_new_valid_exploration(self.EXP_2_ID, self.owner_id)
+        user_models.ExplorationUserDataModel(
+            id='%s.%s' % (self.USER_1_ID, self.EXP_1_ID),
+            user_id=self.USER_1_ID,
+            exploration_id=self.EXP_1_ID,
+        ).put()
+        user_models.ExplorationUserDataModel(
+            id='%s.%s' % (self.USER_2_ID, self.EXP_1_ID),
+            user_id=self.USER_2_ID,
+            exploration_id=self.EXP_1_ID,
+        ).put()
+        user_models.ExplorationUserDataModel(
+            id='%s.%s' % (self.USER_1_ID, self.EXP_2_ID),
+            user_id=self.USER_1_ID,
+            exploration_id=self.EXP_2_ID,
+        ).put()
+
+    def _run_job_and_verify_output(self, expected_output):
+        """Runs the DiscardOldDraftsOneOffJob and verifies that the output
+        matches the expected output.
+
+        Args:
+            expected_output: list(str). The expected output from the one-off
+                job.
+        """
+        job_id = (
+            user_jobs_one_off.DeleteNonExistentExpUserDataJob.create_new()
+        )
+        user_jobs_one_off.DeleteNonExistentExpUserDataJob.enqueue(job_id)
+        self.process_and_flush_pending_mapreduce_tasks()
+        stringified_output = (
+            user_jobs_one_off.DeleteNonExistentExpUserDataJob.get_output(
+                job_id)
+        )
+        eval_output = [
+            ast.literal_eval(stringified_item)
+            for stringified_item in stringified_output
+        ]
+        self.assertItemsEqual(eval_output, expected_output)
+
+    def test_not_deleted_explorations_not_removed(self):
+        rights_manager.publish_exploration(self.owner, self.EXP_1_ID)
+        rights_manager.publish_exploration(self.owner, self.EXP_2_ID)
+
+        self._run_job_and_verify_output([['SUCCESS_KEPT', 3]])
```

Comment:
Here, above, and below: for the everything-succeeds tests we should probably also verify the state of the data store to contrast with the cases where state is being removed.

Reply:
Done.